### ISSUE TITLE
How to navigate from drawer navigator to another with nested navigators ?

### ISSUE THREAD

**Issue by dbyilmaz** [react-navigation/react-navigation#2970]
### Navigator Structure
DrawerNavigators
  - TimelineNavigators
     - Screen1 (initialRoute)
     - Screen2
  - SettingsScreen
### Github Repo
https://github.com/dbyilmaz/drawer
---
### Current Behavior
When click button (onPress={() => { navigation.navigate('Screen2') }) on SettingsScreen, nothing happens. Screen was not changed. is it bug ? 

### Expected Behavior
When click button (onPress={() => { navigation.navigate('Screen2') }) on SettingsScreen , have to open Screen2.

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation | 1.0.0-beta.19
| react-native     | 0.49.5
| node             | 6.11.2
| npm or yarn      | 5.5.1

**Comment by dbyilmaz** [comment]
is it bug? @spencercarli

**Comment by kelset** [comment]
1. Please **never** ping directly maintainers 
1. It may not be a bug and it depends on how you structured your code. Without a Snack or repo reproducing this issue we can't really help.

**Comment by dbyilmaz** [comment]
@kelset 
### TimeLine Navigator
```
export const TimelineNavigators = StackNavigator({
    'screen1': { screen: Timeline, navigationOptions:Screen1NavigatorOptions() },
    'screen2': { screen: LiveMap,  navigationOptions:Screen2NavigatorOptions() },
},{initialRoute: 'screen1'});


```
### Drawer
```
const routeConfiguration = {
    'TimelineNavigators': { screen: TimelineNavigation, navigationOptions: timeLineDrawerOptions },
    'SettingsScreen': { screen: SettingsScreen , navigationOptions: settingsDrawerOptions },
};

export const NavigatorAuth = DrawerNavigator(routeConfiguration, drawerOptions());
```

### Setting Screen

```
import { View, Text, Button } from 'react-native';
import { NavigationActions } from 'react-navigation'

class Settings extends Component {
    render() {
        return (
            <View>
                <Button onPress={()=> this.props.navigation.navigate('screen2')}  title="Go to Screen 2"/>
            </View>
        );
    }
}

export default Settings;

```

I click 'Go to Screen 2 ' button , then not go to Screen 2.

**Comment by dbyilmaz** [comment]
@kelset 

### Setting Screen

```
import { View, Text, Button } from 'react-native';
import { NavigationActions } from 'react-navigation'

class Settings extends Component {
    render() {
        return (
            <View>
                <Button onPress={()=> this.props.navigation.navigate('screen2')}  title="Go to Screen 2"/>
            </View>
        );
    }
}

export default Settings;
```
There is one solution but it is not good.
### Solution:
```
import { View, Text, Button } from 'react-native';
import { NavigationActions } from 'react-navigation'

class Settings extends Component {

//solution
 navigateScreen(navigation){
     navigation.navigate('TimelineNavigators');
     navigation.navigate('screen2');
}
    render() {
        return (
            <View>
                <Button onPress={()=> this.navigateScreen(this.props.navigation)  title="Go to Screen 2"/>
            </View>
        );
    }
}

export default Settings;
```

**Comment by dbyilmaz** [comment]
@kelset  There is more information. I add repo link. Could you help me?

**Comment by dbyilmaz** [comment]
hi @kelset  :) 
is there any improvement?

**Comment by dbyilmaz** [comment]
@kelset

**Comment by kelset** [comment]
> Please never ping directly maintainers

I guess it's really hard to understand. Looks to me like you expect that maintainers are a customer support service, so let le extremely clear:

maintainers !== customer support service

**Comment by dbyilmaz** [comment]
I know @kelset  :)

**Comment by brentvatne** [comment]
this is resolved in 2.0 with https://github.com/react-navigation/rfcs/blob/master/text/0004-less-pushy-navigate.md
